comp/core/configsync/configsyncimpl/sync.go

```diff
@@ -10,12 +10,57 @@ import (
 	"encoding/json"
 	"net/http"
 	"reflect"
+	"strconv"
 	"time"
 
 	apiutil "github.com/DataDog/datadog-agent/pkg/api/util"
 	pkgconfigmodel "github.com/DataDog/datadog-agent/pkg/config/model"
 )
 
+func (cs *configSync) updater() {
+	cfg, err := fetchConfig(cs.ctx, cs.client, cs.Authtoken.Get(), cs.url.String())
+	if err != nil {
+		if cs.connected {
+			cs.Log.Warnf("Failed to fetch config from core agent: %v", err)
+			cs.connected = false
+		} else {
+			cs.Log.Debugf("Failed to fetch config from core agent: %v", err)
+		}
+		return
+	}
+
+	if cs.connected {
+		cs.Log.Debug("Succeeded to fetch config from core agent")
+	} else {
+		cs.Log.Info("Succeeded to fetch config from core agent")
+		cs.connected = true
+	}
+
+	for key, value := range cfg {
+		if key == "logs_config.additional_endpoints" {
+			valueMap, ok := value.(map[string]string)
+			if !ok {
+				updateConfig(cs.Config, key, value)
+			}
+			typedValues := map[string]interface{}{}
+			for cfgkey, cfgval := range valueMap {
+				if cfgkey == "is_reliable" {
+					if b, err := strconv.ParseBool(cfgval); err == nil {
```

Comment:
Why do we have to convert bool here ? Wouldn't the `json.Unmarshal` take care of this ?

As a whole I don't think having post-processing in the sync side of the configuration is a good idea. We just want to mirror configuration, so if the data is not correct we should fix this on the server side not the client. WDYT ?

Reply:
I agree totally with you, post-processing should be minimal/non-existent on the client-side. Currently, because of how `Get()` works and the fact we have nested maps, it becomes very difficult to achieve this as-is. We don't have a `Set()` that will accomplish the same in reverse ensuring the right types (at least it wasn't the case when I tried this simply passing this to `updateConfig`). Thus this sort of post-processing with the map[string]string conversions, and this particular boolean parsing. 

To be honest my fear here is that we might somehow end up with `is_reliable` being marshaled into viper as a `string` and then have some `GetBool()` reference elsewhere maybe break.  

I plan to do follow-up work precisely to make this work as intended - transparently.